examples/08_join_aggregation.py

```diff
@@ -0,0 +1,365 @@
+ratings["timestamp"] = pd.to_datetime(ratings["timestamp"], unit="s")
+
+X = ratings[["userId", "movieId", "timestamp"]]
+y = ratings["rating"]
+X.shape, y.shape
+###############################################################################
+X.head()
+
+###############################################################################
+# Encoding the timestamp with a TableVectorizer
+# ---------------------------------------------
+#
+# Our first step is to extract features from the timestamp, using the
+# |TableVectorizer|. Natively, it uses the |DatetimeEncoder| on datetime
+# columns, and doesn't interact with numerical columns.
+from skrub import TableVectorizer, DatetimeEncoder
+
+
+table_vectorizer = TableVectorizer(
+    datetime_transformer=DatetimeEncoder(add_day_of_the_week=True)
+)
+table_vectorizer.set_output(transform="pandas")
+X_date_encoded = table_vectorizer.fit_transform(X)
+X_date_encoded.head()
+
+###############################################################################
+# We can now make a couple of plots and gain some insight on our dataset.
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+sns.set_style("darkgrid")
+
+
+def make_barplot(x, y, title):
+    norm = plt.Normalize(y.min(), y.max())
+    cmap = plt.get_cmap("magma")
+
+    sns.barplot(x=x, y=y, palette=cmap(norm(y)))
+    plt.title(title)
+    plt.xticks(rotation=30)
+    plt.ylabel(None)
+    plt.tight_layout()
+
+
+# O is Monday, 6 is Sunday
+
+daily_volume = X_date_encoded["timestamp_dayofweek"].value_counts().sort_index()
+
+make_barplot(
+    x=daily_volume.index,
+    y=daily_volume.values,
+    title="Daily volume of ratings",
+)
+
+###############################################################################
+# We also display the distribution of our target ``y``.
+rating_count = y.value_counts().sort_index()
+
+make_barplot(
+    x=rating_count.index,
+    y=rating_count.values,
+    title="Distribution of ratings given to movies",
+)
+
+###############################################################################
+# AggJoiner: aggregate auxiliary tables, then join
+# ------------------------------------------------
+#
+# We now want to extract aggregated datetime features about the users.
+# These features answer questions like
+# *"How many times has this user rated a movie in 2008?"*.
+#
+# Below, |AggJoiner| run the aggregations defined in ``operations`` on the columns
+# ``timestamp_cols`` of the auxiliary table.
+# It does so by grouping by the auxiliary key ``"userId"``, before joining on ``"userId"`` the result
+# back on the main table.
+#
+# Here, the auxiliary table is the table ``X_date_encoded`` itself.
+from skrub import AggJoiner
+
+
+timestamp_cols = [
+    "timestamp_year",
+    "timestamp_month",
+    "timestamp_hour",
+    "timestamp_dayofweek",
+]
+
+agg_joiner_user = AggJoiner(
+    aux_table=X_date_encoded,
+    aux_key="userId",
+    cols=timestamp_cols,
+    main_key="userId",
+    suffix="_user",
```

Comment:
Is specifying the suffix necessary for the example to work (not only here, but all over the example)?

If yes, can we think of a default for the suffix that makes it work here (and is not too strange / magic)

If no, I think that we should remove it from the example, to make the example a bit lighter.

Reply:
This is a relatively complete example. In most simple settings, users won't need to set a suffix.

When a suffix is set, the logic is the following:
1. This suffix is added to each column of the `aux` table after aggregating it.
2. Then, the join procedure is performed between the main table and the aggregated aux table, using the default suffix values, e.g., `("_x", "_y")` for pandas.

Therefore, we won't have errors if we don't set the suffix in this example. However, we will have columns suffixed with variations of `_x` and `_y`, which will be hard to decipher.

Note that if we input severable tables without setting suffixes, we will automatically generate suffixes using the index of the auxiliary tables (`_1`, `_2`, ...).

However, if we call several `AggJoiner` like in this example, I'm afraid we can't make good default suffixes.

WDYT?

Reply:
I think the meaningful suffixes "_movie" and "_user" actually help understand what we are doing and what we see in the cells' outputs (although that part is a bit hard still because the tables are quite wide)